### ISSUE TITLE
[question] CMakeDeps: why True/False options are translated to CACHE STRING variables and not CACHE BOOL?

### ISSUE THREAD

**Issue by SpaceIm** [conan-io/conan#11853]
- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

Let's consider a recipe like this:

```python
from conan import ConanFile
from conan.tools.cmake import CMake, CMakeToolchain


class FooConan(ConanFile):
    (...)

    options = {
        bar: [True, False],
    }
    default_options = {
        bar: True,
    }

    (...)

    def generate(self):
        tc = CMakeToolchain(self)
        tc.variables["OPTION1"] = False
        tc.variables["OPTION2"] = self.options.bar
        tc.generate()

    (...)
```

In `conan_toolchain.cmake`, you'll see:

```cmake
set(OPTION1 OFF CACHE BOOL "Variable OPTION1 conan-toolchain defined")
set(OPTION2 "True" CACHE STRING "Variable OPTION2 conan-toolchain defined")
```

Is it intended? Is it robust? Was it deeply tested (various cmake_minimum_required, some if() based on this option in a CMakeLists etc)? I've tested, and if(OPTION2) seems to work fine, but I'm always suspicious with variable expansion in CMake.

**Comment by jens-f** [comment]
We ran into this as well and it seems to be working despite the string value. The cause might be an issue with Python not automatically converting the option values into "real" True/False values when applying it into the variables dictionary. The Conan package options class even has the `__bool__` function but it just doesn't seem to be called automatically in this case. In order to force this you need to call
```
tc.variables["OPTION2"] = bool(self.options.bar)
```

**Comment by kambala-decapitator** [comment]
duplicate of #14590, should be closed

**Comment by memsharded** [comment]
Yes, duplicate and resolved by comment https://github.com/conan-io/conan/issues/14590#issuecomment-2059439834, closing
